devon4j-migrate/index.asciidoc

```diff
@@ -0,0 +1,51 @@
+= Migrate a devon4j project to the latest version
+====
+This tutorial mainly focuses on migrating an older version of devon4j project to the latest version using devon4j migrate command.
+
+
+## Prerequisites
+* devonfw IDE
+
+## Learning goals.
+Here in this tutorial you will learn 
+* How to migrate an older version of devon4j project to the latest version.
+====
+
+[step]
+--
+restoreDevonfwIde(["java","vscode"])
+--
+
+
+[step]
+--
+restoreWorkspace({"workspace": "devon4j-migrate"})
+--
+
+====
+To demonstrate devon4j migrate you would need a sample devon4j project with any older version. Here, we have already created a project with an older version i.e. 2020.08.001 for you. 
+
+It is already cloned in workspace using restoreWorkspace.
+
+Execute below command for project migration from an older version to the latest version.
+[step]
+== Execute Migrate Command
+--
+executeCommand("devon java migrate", "devon java migrate", {"dir" : "devonfw/workspaces/main"})

```

Comment:
working directory is already ../devonfw/workspaces/main. in this case you would run this in ../devonfw/workspaces/main/devonfw/workspaces/main

Reply:
Are you sure about that? In katacoda you need to change the dir. Do we have an error here?

Reply:
Hi Denise, If I keep only the name of project then I have to manually navigate to  "devonfw/workspaces/main" from command line in katacoda environment and then execute command for directory i.e. "devon4j-app-4tutorial" which I have added. then I have to execute migrate command

Reply:
Yes, when we start restoreWorkspace, we only clone the git repository into the workspaces folder and set the working directory to root/devonfw/workspaces/main. Since we haven't changed the folders in katacoda yet, we would still be in the root-folder when the tutorial starts. That's why we have an cd command at the beginning. While writing the tutorial on the other hand, we would have to specify the parameter relative to the workspace directory, which is already root/devonfw/workspaces/main.

Reply:
We can change that if it's to confusing this way. But since we usually mean by 'relative to workspace', relative to main, we kept it that way

Reply:
There is actually a bug in executeCommand. It only returns the path not the relative path. I will fix it

Reply:
Thank you Denise for the fix. It is working fine now